### ISSUE TITLE
[Suggestion] Don't require test module/log "no tests" for a package with tests in a separate package

### ISSUE THREAD

**Issue by helmesjo** [build2/build2#192]
Following the [recommended approach](https://github.com/build2/HOWTO/blob/master/entries/handle-tests-with-extra-dependencies.md) on how to setup tests that require additional imports, I noticed a few oddities (when everything is setup):
1. Package `libhello` is required to import the test module (`using test`) in it's `bootstrap.build` file, else then running `bdep test` you see this:
    `<buildspec>: error: target ../hello-clang13/hello/ does not support operation test`
2. If `using test` is let be, each time running `bdep test` you'll get this:
    `info: ../hello-clang13/dir{hello/} has nothing to test`

This seems a bit counter-intuitive (and confusing) since this whole constellation is for saying that `libhello` _has_ tests, but they are in package `libhello-tests`". 

**Suggestions**
1. Could `bdep test` recognize that this is the case, and when it sees `libhello` just go straight to `libhello-tests` and run those (or if they've already been run then do nothing, I don't now the execution order of `bdep test`)? So that one can (if desired) remove `using test` in `libhello` (since, again, per se it doesn't require the `test` module so one would assume it'll work to not specify it).
2. If not, perhaps not print the info log in this case, since (I'm assuming) it should see that if you want to test `libhello`, the tests are in fact in `libhello-tests` (unless `libhello` itself adds additional tests for some reason, eg. testscript things).

The main thing here is to reduce confusion, but also "friction" when working with a project. And in that spirit is the desire to easily just run `bdep test` to "run everything testable" and not error out on things that say that they don't have tests.

Just some feedback! 
Thanks.

**Comment by boris-kolpackov** [comment]
Yes, there is a way (via `b info`) to detect whether a project supports the `test` operation. We already do this in CI and should also do in `bdep` (and/or `bpkg`).
